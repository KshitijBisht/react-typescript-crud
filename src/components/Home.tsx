import * as React from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import axios from 'axios';
import {RootState} from '../store/modules/combineReducers';
import { Customer } from '../store/modules/customer/types';
import {deleteCustomer} from '../store/modules/customer/actions'

export default function Home(props:RouteComponentProps){
    const dispatch = useDispatch();
    // const [customers,setCustomers] = React.useState<any[]>([])


    // const fetchCustomers = function(){
    //     axios.get(`http://localhost:5000/customers`).then(data => setCustomers(data.data))
    // }

    // React.useEffect(() => {
    //     fetchCustomers();
    //    }, [])

    const customers = useSelector((state: RootState) => state.customer.data);

       const handleDelete = function(customer:Customer,
        e: React.MouseEvent<HTMLSpanElement, MouseEvent>
        ): void {
            e.stopPropagation();
            const answer = window.confirm('Do you really want to delete this entry ?')
           if(answer){
               dispatch(deleteCustomer(customer))
        //     axios.delete(`http://localhost:5000/customers/${id}`).then(data => {
        //     const index = customers.findIndex(customer => customer.id === id);
        //     customers.splice(index, 1);
        //     props.history.push('/');
        // })

           }
           
    }
        return (
            <div>
                {customers.length === 0 && (
                    <div className="text-center">
                        <h2>No Users found at the moment</h2>
                    </div>
                )}
                <div className="container">
                    <div className="row">
                        <table className="table table-striped">
                            <thead className="thead-light">
                                <tr>
                                    <th scope="col">Name</th>
                                    <th scope="col">Description</th>
                                    <th scope="col">Updated By</th>
                                    <th scope="col">Last Updated</th>
                                    <th scope="col">Actions</th>

                                </tr>
                            </thead>
                            <tbody>
                                {customers && customers.map(customer =>
                                    <tr key={customer.id}>
                                        <td><Link to={`edit/${customer.id}`}>{customer.name}</Link></td>
                                        <td>{customer.description}</td>
                                        <td>{customer.updated_by}</td>
                                        <td>{customer.last_updated}</td>
                                        <td>
                                            <div className="d-flex justify-content-between align-items-center">
                                                <div className="btn-group" style={{ marginBottom: "20px" }}>
                                                <Link to={`edit/${customer.id}`}><i className ="fa fa-pencil" style={{marginRight: "10px"}}></i></Link>
                                                    <i className="fa fa-trash" id="delete-icon" onClick={(e): void => handleDelete(customer,e)} ></i>
                                                </div>
                                            </div>
                                        </td>
                                    </tr>
                                )}
                            </tbody>
                        </table>
                    </div>
                </div>
            </div>
        )
    
}