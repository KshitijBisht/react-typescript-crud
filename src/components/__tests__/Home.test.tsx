import Home from '../Home'
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import shallowToJson from 'enzyme-to-json';
import { createMemoryHistory, createLocation } from 'history';
import { match } from 'react-router';

const history = createMemoryHistory();
const path = `/route/:id`;

const match: match<{ id: string }> = {
    isExact: false,
    path,
    url: path.replace(':id', '1'),
    params: { id: "1" }
};

const location = createLocation(match.url);

Enzyme.configure({ adapter: new Adapter() });

describe('<Home/>',()=>{
    const wrapper= shallow(<Home history={history} location={location} match={match}/>);
    it('should be defined',()=>{
    expect(wrapper).toBeDefined();
    })
  
    it('should match the snapshot', () => {
          expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

    //   it('should render deleteCustomer icon',()=>{
    //       const component = shallow(<Home history={history} location={location} match={match}/>)
    //       const instance = component.instance();
    //       instance.deleteCustomer();
    //   });.
  });