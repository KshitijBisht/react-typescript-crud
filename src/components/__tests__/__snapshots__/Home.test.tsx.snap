// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should match the snapshot 1`] = `
<div>
  <div
    className="text-center"
  >
    <h2>
      No Users found at the moment
    </h2>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <table
        className="table table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr>
            <th
              scope="col"
            >
              Name
            </th>
            <th
              scope="col"
            >
              Description
            </th>
            <th
              scope="col"
            >
              Updated By
            </th>
            <th
              scope="col"
            >
              Last Updated
            </th>
            <th
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
